src/MinterGateway.sol

```diff
@@ -794,8 +770,28 @@ contract MinterGateway is IMinterGateway, ContinuousIndexing, ERC712Extended {
 
         if (principalOfMaxAllowedActiveOwedM_ >= principalOfActiveOwedM_) return;
 
+        MinterState storage minterState_ = _minterStates[minter_];
+
+        uint40 penalizeFrom_ = UIntMath.max40(minterState_.updateTimestamp, minterState_.penalizedUntilTimestamp);
+
+        if (newTimestamp_ <= penalizeFrom_) return;
+
+        uint32 updateCollateralInterval_ = updateCollateralInterval();
+
+        // NOTE: If `updateCollateralInterval_` is zero, then we can assume the minter would never call
+        //       `updateCollateral` (which is the only caller of this function) until they have collateral to report
+        //       that would not result in undercollaterilization penalties, so skip charging these penalties.
+        if (updateCollateralInterval_ == 0) return;
```

Comment:
This check should live at the top of the function.

Reply:
It cannot because `updateCollateralInterval()` is only fetched as late as possible, as it is the least likely to be 0.